pkg/ingester/ingester.go

```diff
@@ -1278,8 +1270,9 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 	return nil
 }
 
-func (i *Ingester) QueryExemplars(ctx context.Context, req *client.ExemplarQueryRequest) (*client.ExemplarQueryResponse, error) {
-	if err := i.checkRunning(); err != nil {
+func (i *Ingester) QueryExemplars(ctx context.Context, req *client.ExemplarQueryRequest) (resp *client.ExemplarQueryResponse, qeErr error) {
+	defer func() { qeErr = i.handleReadError(qeErr) }()
```

Comment:
Same comment as before, i.e. I think that the returned error should be named `err` to avoid confusion and potential programming mistakes.